Add common scenario for commit failure (#7433)
pingcap/docs: sql-statements/sql-statement-commit.md

## MySQL compatibility

> * TiDB does not currently use Metadata Locking (MDL) to prevent DDL statements from modifying tables that are in use by transactions. In the event that table a definition has changed, committing a transaction will result in an `Information schema is changed` error. In this event, the transaction is rolled back automatically.

Reviewer: ```suggestion
* Currently, TiDB does not use Metadata Locking (MDL) to prevent DDL statements from modifying tables used by transactions. If the definition of a table has changed, committing a transaction will result in an `Information schema is changed` error. In this event, the transaction is rolled back automatically.
```
What does "table a definition" mean?

Author: It should just be "table definition". It means the columns, names of the columns etc.
